Fix TypeError with benchmark-rule script

<!-- Please answer the following. Issues that do not will be closed. -->

> Clearly describe the bug

When running the command in [the documentation](https://github.com/stylelint/stylelint/blob/f865f3df5d67f97aa2d694d56f81d1ba5613f8b7/docs/developer-guide/rules.md#improving-the-performance-of-a-new-or-an-existing-rule), the following error is raised:

```
$ npm run benchmark-rule -- selector-combinator-space-after never

> stylelint@9.8.0 benchmark-rule /Users/koba/git/stylelint/stylelint
> node scripts/benchmark-rule.js "selector-combinator-space-after" "never"

/Users/koba/git/stylelint/stylelint/scripts/benchmark-rule.js:34
const rule = rules[ruleName].apply(null, normalizeRuleSettings(parsedOptions));
                             ^

TypeError: Cannot read property 'apply' of undefined
    at Object.<anonymous> (/Users/koba/git/stylelint/stylelint/scripts/benchmark-rule.js:34:30)
    at Module._compile (internal/modules/cjs/loader.js:707:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:718:10)
    at Module.load (internal/modules/cjs/loader.js:605:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:544:12)
    at Function.Module._load (internal/modules/cjs/loader.js:536:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:760:12)
    at startup (internal/bootstrap/node.js:308:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:878:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! stylelint@9.8.0 benchmark-rule: `node scripts/benchmark-rule.js "selector-combinator-space-after" "never"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the stylelint@9.8.0 benchmark-rule script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/koba/.npm/_logs/2018-11-12T15_37_35_699Z-debug.log
```

> Which rule, if any, is the bug related to?

There may be something wrong in `scripts/benchmark-rule.js`.

https://github.com/stylelint/stylelint/blob/f865f3df5d67f97aa2d694d56f81d1ba5613f8b7/scripts/benchmark-rule.js#L34

> What CSS is needed to reproduce the bug?

N/A

> What stylelint configuration is needed to reproduce the bug?

N/A

> Which version of stylelint are you using?

`9.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI script for developers.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

N/A

> What did you expect to happen?

- Benchmark for all existing rules succeed.
- All examples of the benchmark script in documentation are fixed.

> What actually happened (e.g. what warnings or errors did you get)?

`npm run benchmark-rule` command failed.

--- a/docs/developer-guide/rules.md
+++ b/docs/developer-guide/rules.md
@@ -283,10 +283,6 @@ npm run benchmark-rule -- selector-combinator-space-after always
 ```
 
 ```shell
-npm run benchmark-rule -- selector-no-combinator true
-```
-
-```shell
 npm run benchmark-rule -- block-opening-brace-space-before "[\"always\", {\"ignoreAtRules\": [\"else\"]}]"
 ```
 

--- a/docs/developer-guide/rules.md
+++ b/docs/developer-guide/rules.md
@@ -269,10 +269,10 @@ Deprecating rules doesn't happen very often. However, these two steps are import
 There's a simple way to run benchmarks on any given rule with any valid config for it:
 
 ```shell
-npm run benchmark-rule -- [rule-name] [config]
+npm run benchmark-rule -- ruleName ruleOptions [ruleContext]
 ```
 
-If the `config` argument is anything other than a string or a boolean, it must be valid JSON wrapped in quotation marks.
+If the `ruleOptions` argument is anything other than a string or a boolean, it must be valid JSON wrapped in quotation marks.
 
 ```shell
 npm run benchmark-rule -- selector-combinator-space-after never
@@ -284,6 +284,12 @@ npm run benchmark-rule -- selector-combinator-space-after always
 
 ```shell
 npm run benchmark-rule -- block-opening-brace-space-before "[\"always\", {\"ignoreAtRules\": [\"else\"]}]"
+```
+
+If the `ruleContext` argument is specified, the sames procedure would apply:
+
+```shell
+npm run benchmark-rule -- block-opening-brace-space-before "[\"always\", {\"ignoreAtRules\": [\"else\"]}]" "{\"fix\": \"true\"}"
 ```
 
 The script loads Bootstrap's CSS (from its CDN) and runs it through the configured rule.

--- a/scripts/benchmark-rule.js
+++ b/scripts/benchmark-rule.js
@@ -69,7 +69,7 @@ let firstTime = true;
 
 function benchFn(css, done) {
   processor
-    .process(css)
+    .process(css, { from: undefined })
     .then(result => {
       if (firstTime) {
         firstTime = false;

--- a/scripts/benchmark-rule.js
+++ b/scripts/benchmark-rule.js
@@ -7,13 +7,14 @@ const normalizeRuleSettings = require("../lib/normalizeRuleSettings");
 const postcss = require("postcss");
 const request = require("request");
 const requireRule = require("../lib/requireRule");
-const rules = require("../lib/rules");
 
 const ruleName = process.argv[2];
 const ruleOptions = process.argv[3];
 const ruleContext = process.argv[4];
 
-if (!rules.includes(ruleName)) {
+const ruleFunc = requireRule(ruleName);
+
+if (!ruleFunc) {
   throw new Error("You must specify a valid rule name");
 }
 
@@ -43,7 +44,7 @@ const primary = ruleSettings[0];
 const secondary = ruleSettings[1] || null;
 const context = ruleContext ? JSON.parse(ruleContext) : {};
 
-const rule = requireRule(ruleName).apply(null, [primary, secondary, context]);
+const rule = ruleFunc.apply(null, [primary, secondary, context]);
 
 const processor = postcss().use(rule);
 

--- a/scripts/benchmark-rule.js
+++ b/scripts/benchmark-rule.js
@@ -6,13 +6,19 @@ const chalk = require("chalk");
 const normalizeRuleSettings = require("../lib/normalizeRuleSettings");
 const postcss = require("postcss");
 const request = require("request");
+const requireRule = require("../lib/requireRule");
 const rules = require("../lib/rules");
 
 const ruleName = process.argv[2];
 const ruleOptions = process.argv[3];
+const ruleContext = process.argv[4];
 
-if (!ruleName || !ruleOptions) {
-  throw new Error("You must specify a rule name and rule options");
+if (!rules.includes(ruleName)) {
+  throw new Error("You must specify a valid rule name");
+}
+
+if (!ruleOptions) {
+  throw new Error("You must specify rule options");
 }
 
 const CSS_URL =
@@ -31,7 +37,14 @@ if (
 }
 
 /* eslint-enable eqeqeq */
-const rule = rules[ruleName].apply(null, normalizeRuleSettings(parsedOptions));
+const ruleSettings = normalizeRuleSettings(parsedOptions);
+
+const primary = ruleSettings[0];
+const secondary = ruleSettings[1] || null;
+const context = ruleContext ? JSON.parse(ruleContext) : {};
+
+const rule = requireRule(ruleName).apply(null, [primary, secondary, context]);
+
 const processor = postcss().use(rule);
 
 request(CSS_URL, (error, response, body) => {

